salt/grains/core.py

```diff
@@ -1788,8 +1786,12 @@ def os_data():
         # so that linux_distribution() does the /etc/lsb-release parsing, but
         # we do it anyway here for the sake for full portability.
         if 'osfullname' not in grains:
-            grains['osfullname'] = \
-                grains.get('lsb_distrib_id', osname).strip()
+            # If NI Linux RT distribution, set the grains['osfullname'] to 'nilrt'
+            if grains.get('lsb_distrib_id', '').startswith('nilrt'):
```

Comment:
Are you sure this will always startwith a _lowercase_ `nilrt`? Most distribution IDs are not all lowercase. Maybe it would make sense to use `.lower()` here.

```python
if grains.get('lsb_distrib_id', '').lower().startswith('nilrt'):
```

Reply:
Okay, that makes sense! NI's linux distribution relies on OpenEmbedded, and that's lower-case based, but it doesn't hurt comparing with lower() case.

Done!